sdk/eventhub/event-hubs/src/managementClient.ts

```diff
@@ -306,31 +309,132 @@ export class ManagementClient extends LinkEntity {
+      const aborter: AbortSignalLike | undefined = options && options.abortSignal;
+
+      const sendOperationPromise = () =>
+        new Promise<Message>(async (resolve, reject) => {
+          try {
+            if (!options) {
+              options = {};
+            }
+
+            let timeOver: boolean = false;
+            const retryTimeoutInMs = getRetryAttemptTimeoutInMs(options.retryOptions);
+            let timeTakenByInit = 0;
+
+            if (!this._isMgmtRequestResponseLinkOpen()) {
+              log.mgmt(
+                "[%s] Acquiring lock to get the management req res link.",
+                this._context.connectionId
+              );
+
+              const rejectOnAbort = () => {
+                const address = this._mgmtReqResLink!.receiver.address || "address";
+                const requestName = options!.requestName;
+                const desc: string =
+                  `[${this._context.connectionId}] The request "${requestName}" ` +
+                  `to "${address}" has been cancelled by the user.`;
+                log.error(desc);
+                const error = new AbortError(
+                  `The ${
+                    requestName ? requestName + " " : ""
+                  }operation has been cancelled by the user.`
+                );
+
+                return reject(error);
+              };
+
+              const onAbort = () => {
+                // safe to clear the timeout if it hasn't already occurred.
+                if (!timeOver) {
+                  clearTimeout(waitTimer);
+                }
+                aborter!.removeEventListener("abort", onAbort);
+
+                rejectOnAbort();
+              };
+
+              if (aborter) {
+                if (aborter.aborted) {
+                  return rejectOnAbort();
+                }
+                aborter.addEventListener("abort", onAbort);
+              }
+
+              const initOperationStartTime = Date.now();
+
+              const actionAfterTimeout = () => {
+                timeOver = true;
+                const address = this._mgmtReqResLink!.receiver.address || "address";
+                const desc: string =
+                  `The request with message_id "${request.message_id}" to "${address}" ` +
+                  `endpoint timed out. Please try again later.`;
+                const e: Error = {
+                  name: "OperationTimeoutError",
+                  message: desc
+                };
+                return reject(translate(e));
+              };
+
+              const waitTimer = setTimeout(actionAfterTimeout, retryTimeoutInMs);
+
+              try {
+                await defaultLock.acquire(this.managementLock, () => {
+                  return this._init();
+                });
+              } catch (err) {
+                return reject(translate(err));
+              } finally {
+                clearTimeout(waitTimer);
+                aborter!.removeEventListener("abort", onAbort);
```

Comment:
aborter can be undefined

```suggestion
if (aborter) {
        aborter.removeEventListener("abort", onAbort);
 }
```